plugins/versioning/src/main/kotlin/versioning/VersionsNavigationCreator.kt

```diff
@@ -1,53 +1,84 @@
+        if (versions.size == 1) {
+            return versioningStorage.currentVersion.name
+        }
         val position = output.takeIf { it.isDirectory } ?: output.parentFile
-        return versioningHandler.getVersions()
+        if(isOnlyOnRootPage ) {
+            getActiveVersion(position)?.takeIf {
+                it.value == versioningStorage.currentVersion.dir
+                        && it.value != position
+            }?.also { return@invoke it.key }
+        }
+        return versions
             .let { versions -> versionsOrdering.order(versions.keys.toList()).map { it to versions[it] } }
             .takeIf { it.isNotEmpty() }
-            ?.let { versions ->
+            ?.let { orderedVersions ->
                 StringBuilder().appendHTML().div(classes = "versions-dropdown") {
-                    button(classes = "versions-dropdown-button") {
-                        versions.first { (_, versionLocation) -> versionLocation?.absolutePath == position.absolutePath }
-                            .let { (version, _) ->
-                                text(version)
-                            }
-                        i(classes = "fa fa-caret-down")
+                    val activeVersion = getActiveVersion(position)
+                    val relativePosition: String  = activeVersion?.value?.let { output.toRelativeString(it) } ?: "index.html"
+                    div(classes = "versions-dropdown-button") {
+                        activeVersion?.key?.let { text(it) }
                     }
                     div(classes = "versions-dropdown-data") {
-                        versions.forEach { (version, path) ->
-                            a(href = path?.resolve("index.html")?.toRelativeString(position)) {
-                                text(version)
+                        orderedVersions.forEach { (version, path) ->
+                            if (version == activeVersion?.key) {
+                                a(href = output.name) { text(version) }
+                            } else {
+                                val isExistsFile =
+                                    if (version == versioningStorage.currentVersion.name)
+                                        path?.resolve(relativePosition)?.exists() == true
+                                    else
+                                        versioningStorage.previousVersions[version]?.src?.resolve(relativePosition)
+                                            ?.exists() == true
+
+                                val absolutePath =
+                                    if(isExistsFile) path?.resolve(relativePosition) else path?.resolve("index.html")
+
+                                a(href = absolutePath?.toRelativeString(position)) {
+                                    if (isExistsFile)
+                                        text(version)
+                                    else
+                                        strike { text(version) }
```

Comment:
not sure if everyone will know why some versions are crossed out. Can we have some tooltip or any other form of informing documentation user that this is not present in certain version. Maybe Eugene / Mike will have a better idea for that